src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.Layouting.Arrange.cs

```diff
@@ -0,0 +1,324 @@
+using System;
+using System.Runtime.CompilerServices;
+using Windows.Foundation;
+
+#if IS_WINUI
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+#else
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+#endif
+
+namespace Uno.Toolkit.UI;
+
+partial class AutoLayout
+{
+	protected override Size ArrangeOverride(Size finalSize)
+	{
+		// Local cache of DependencyProperties
+		var children = Children;
+		var orientation = Orientation;
+		var justify = Justify;
+		var primaryAxisAlignment = PrimaryAxisAlignment;
+		var padding = Padding;
+		var spacing = Spacing is > double.NegativeInfinity and < double.PositiveInfinity and { } s ? s : 0d;
```

Comment:
what about `var spacing = Spacing.FiniteOrDefault(defaultValue: 0d)` ?

Reply:
We're not familiar with existing extensions in the project... but yeah, looks like it's doing the same job.